ownCloud/Client/ClientRootViewController.swift

```diff
@@ -155,32 +166,28 @@ class ClientRootViewController: UITabBarController {
 
 		self.tabBar.applyThemeCollection(Theme.shared.activeCollection)
 
-		self.viewControllers = [filesNavigationController] as? [UIViewController]
-	}
-
-	func logoutBarButtonItem() -> UIBarButtonItem {
-		let barButton = UIBarButtonItem(title: "Disconnect".localized, style: .plain, target: self, action: #selector(logout(_:)))
-		barButton.accessibilityIdentifier = "disconnect-button"
-		return barButton
-	}
-
-	@objc func logout(_: Any) {
-		self.closeClient()
+        self.viewControllers = [filesNavigationController] as? [UIViewController]
 	}
 
 	func closeClient(completion: (() -> Void)? = nil) {
-		self.presentingViewController?.dismiss(animated: true, completion: completion)
+        self.navigationController?.setNavigationBarHidden(false, animated: false)
+        self.navigationController?.popViewController(animated: true, completion: {
```

Comment:
@hosy is `completion` lambda not optional here?

Reply:
That's right!